src/Nethermind/Nethermind.Evm/BlobGasCalculator.cs

```diff
@@ -50,35 +50,40 @@ public static bool TryCalculateFeePerBlobGas(ulong excessBlobGas, UInt256 blobGa
     {
         static bool FakeExponentialOverflow(UInt256 factor, UInt256 num, UInt256 denominator, out UInt256 feePerBlobGas)
         {
-            UInt256 output = UInt256.Zero;
-
-            if (UInt256.MultiplyOverflow(factor, denominator, out UInt256 numAccum))
+            UInt256 accumulator;
+            if (factor == UInt256.One)
+            {
+                // Skip expensive 256bit multiplication if factor is 1
+                accumulator = denominator;
+            }
+            else if (UInt256.MultiplyOverflow(factor, denominator, out accumulator))
             {
                 feePerBlobGas = UInt256.MaxValue;
                 return true;
             }
 
-            for (UInt256 i = 1; numAccum > 0; i++)
+            UInt256 output = default;
+            for (ulong i = 1; !accumulator.IsZero; i++)
```

Comment:
in practice accumulator should never go pass `ulong`, should we have separate path for it?